packages/next/src/server/use-cache/use-cache-wrapper.ts

```diff
@@ -495,6 +502,31 @@ export function cache(kind: string, id: string, fn: any) {
         }
       }
 
+      if (boundArgsLength > 0) {
+        if (args.length === 0) {
+          throw new InvariantError(
+            `Expected the "use cache" function ${JSON.stringify(fn.name)} to receive its encrypted bound arguments as the first argument.`
+          )
+        }
+
+        const [encryptedBoundArgs, ...otherArgs] = args
+        const boundArgs = await decryptActionBoundArgs(id, encryptedBoundArgs)
+
+        if (!Array.isArray(boundArgs)) {
+          throw new InvariantError(
+            `Expected the bound arguments of "use cache" function ${JSON.stringify(fn.name)} to deserialize into an array, got ${typeof boundArgs} instead.`
+          )
+        }
+
+        if (boundArgsLength !== boundArgs.length) {
+          throw new InvariantError(
+            `Expected the "use cache" function ${JSON.stringify(fn.name)} to receive ${boundArgsLength} bound arguments, got ${boundArgs.length} instead.`
+          )
+        }
+
+        args = [...boundArgs, ...otherArgs]
```

Comment:
That's a lot of copying. maybe just loop over args from index 1 and push onto boundArgs

Reply:
i mean, both of these are probably unlikely to be more than 5-10 elements, does it rly matter

Reply:
In this case probably not super meaningful but next is not fast and to make it fast we need to be considerate of unessecary overhead. More so in hot paths which arguably this is not one but at the same time it’s more about building habits that pay off over the long term as hundreds and thousands of these small optimizations add up